clients.go

```diff
@@ -356,6 +357,29 @@ func (cl *Client) Read(packetHandler ReadFn) error {
 	}
 }
 
+// openShutdownSignal initializes a shutdown signal channel for the client.
+// This channel is used to wait for a shutdown notification.
+func (cl *Client) openShutdownSignal() {
+	cl.Lock()
+	defer cl.Unlock()
+	cl.State.shutdownSignal = make(chan any)
+}
+
+// waitForShutdownSignal waits for the shutdown signal to be received.
+func (cl *Client) waitForShutdownSignal() {
+	if cl.State.shutdownSignal != nil {
+		<-cl.State.shutdownSignal
+	}
+}
```

Comment:
If cl.State.shutdownSignal is initialized in NewClient, we can remove if check.

Reply:
Refer to the previous comment.